src/index.ts

```diff
@@ -125,9 +125,10 @@ export async function main(): Promise<void> {
 
     let dockerRegistryClient: DockerRegistryClient | null = null;
     let finalizeDockerRegistryClient: (() => Promise<void>) | null = null;
-    const artifactRegistryRepository =
+    const artifactRegistryRepository = encodeURIComponent(
```

Comment:
Is this the right spot? I think this is where the action reads the top level repository from the github workflows config, not the individual docker image from the application-values.yaml file. (It's incredibly confusing that AR and Docker use "repository" to mean different things.)

Reply:
we're you thinking it should go somewhere within the ArtifactRegistryDockerRegistryClient class itself?